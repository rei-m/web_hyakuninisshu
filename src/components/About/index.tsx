import * as React from 'react';

const About = () => {
  return <div>サイトについて</div>;
};

export default About;
