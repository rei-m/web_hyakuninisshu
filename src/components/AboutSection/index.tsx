import * as React from 'react';
import styled from 'styled-components';
import { withTheme } from '../../styles';

const Root = withTheme(styled.div)`
  padding: ${({ theme }) => theme.spacing2x};
  box-sizing: border-box;
  max-width: 960px;
  margin: auto;
  text-align: left;
`;

const Section = withTheme(styled.section)`
  margin: ${({ theme }) => theme.spacing4x} 0;
`;

const SectionTitle = styled.h1`
  font-size: 2.4rem;
  position: relative;
  &:after {
    content: '';
    width: 100%;
    border-bottom: 1px solid #a9a9a9;
    position: absolute;
    bottom: -8px;
    left: 0;
  }
`;

const SectionText = withTheme(styled.div)`
  padding: ${({ theme }) => theme.spacing1x};
  line-height: 2rem;
`;

const AppBannerBox = styled.div`
  text-align: center;
`;

const AboutSection = () => (
  <Root>
    <Section>
      <SectionTitle>サイトについて</SectionTitle>
      <SectionText>
        このサイトは百人一首を覚えるためのサイトです。4択のクイズ形式で繰り返し練習することで効率よく百人一首を覚えることが出来ます。
      </SectionText>
    </Section>
    <Section>
      <SectionTitle>使い方について</SectionTitle>
      <SectionText>
        練習では様々な条件を組み合わせて出題範囲を指定できます。ご自身の習熟度に合わせて効率よく練習できます。
        もう全部覚えたという方は腕試しに挑戦しましょう。百首通しでランダムに出題されます。
        資料では百人一首の詳細な情報を閲覧できます。決まり字や歌の現代語訳などを確認することができます。
      </SectionText>
    </Section>
    <Section>
      <SectionTitle>アプリ版について</SectionTitle>
      <SectionText>
        Androidのみリリースしています。こちらは一旦インストールしていただけばオフラインでも使えます。iOS版は未定ですが、要望があればがんばります。
      </SectionText>
      <AppBannerBox>
        <a href="https://play.google.com/store/apps/details?id=me.rei_m.hyakuninisshu&hl=ja&pcampaignid=MKT-Other-global-all-co-prtnr-py-PartBadge-Mar2515-1">
          <img
            alt="Google Play で手に入れよう"
            src="https://play.google.com/intl/en_us/badges/images/generic/ja_badge_web_generic.png"
            style={{
              margin: 'auto',
              width: 200
            }}
          />
        </a>
      </AppBannerBox>
    </Section>
    <Section>
      <SectionTitle>運営者について</SectionTitle>
      <SectionText>
        当サイトは
        <a href="https://twitter.com/rei_m" target="_blank">
          @rei_m
        </a>
        が個人で運営しています。不具合や要望等あればTwitterまでご連絡ください。
        アプリのレビューで読み上げ機能が欲しいというお声をいただくのですが、使用可能ないい音源があれば搭載します。
      </SectionText>
    </Section>
  </Root>
);

export default AboutSection;
