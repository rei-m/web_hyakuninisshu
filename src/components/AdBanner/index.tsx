import * as React from 'react';
import AdSense from 'react-adsense';
import styled from '@src/styles/styled-components';

const Container = styled.div`
  padding: ${({ theme }) => theme.spacing2x};
`;

const AdBanner: React.FC<{}> = () => (
  <Container>
    <AdSense.Google
      client="ca-pub-4104372369598017"
      slot="9171068817"
      style={{ display: 'block' }}
      format="auto"
      responsive="true"
    />
  </Container>
);

export default AdBanner;
