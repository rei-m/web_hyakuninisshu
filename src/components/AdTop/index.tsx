import * as React from 'react';
import AdSense from 'react-adsense';
import styled from '@src/styles/styled-components';

const Container = styled.div`
  text-align: center;
  padding: ${({ theme }) => theme.spacing1x} 0;
`;

const Dummy = styled.div`
  width: 320px;
  height: 100px;
  border: 1px solid #123;
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 0 auto;
  @media (min-width: 500px) {
    width: 468px;
    height: 60px;
  }
  @media (min-width: 800px) {
    width: 728px;
    height: 90px;
  }
`;

const AdTop: React.FC<{ style?: React.CSSProperties }> = ({ style }) =>
  process.env.NODE_ENV === 'production' ? (
    <Container style={style}>
      <style
        dangerouslySetInnerHTML={{
          __html: `
  .ad_top { width: 320px; height: 100px; }
  @media(min-width: 500px) { .ad_top { width: 468px; height: 60px; } }
  @media(min-width: 800px) { .ad_top { width: 728px; height: 90px; } }　        
`,
        }}
      />
      <AdSense.Google
        className="ad_top"
        client="ca-pub-4104372369598017"
        slot="9171068817"
        style={{ display: 'inline-block' }}
        format=""
        responsive=""
      />
    </Container>
  ) : (
    <Container style={style}>
      <Dummy>この欄は広告枠です</Dummy>
    </Container>
  );

export default AdTop;
