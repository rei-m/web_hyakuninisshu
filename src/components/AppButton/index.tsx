import * as React from 'react';
import withStyles from '@material-ui/core/styles/withStyles';
import Icon from '@material-ui/core/Icon';
import Button from '@material-ui/core/Button';
import { appTheme } from '@src/styles/theme';

export interface Props {
  label: string;
  type: 'primary' | 'normal';
  icon?: string;
  style?: React.CSSProperties;
  onClick: () => void;
}

const styles = {
  button: {
    padding: '4px 15px',
    fontSize: '1.6rem',
    height: '40px',
  },
  normalButton: {
    color: '#182026',
    backgroundColor: '#f5f8fa',
    '&:active': {
      backgroundColor: appTheme.colorAccentActive,
    },
    '&:hover': {
      backgroundColor: '#ebf1f5',
    },
  },
  primaryButton: {
    color: '#ffffff',
    backgroundColor: appTheme.colorAccent,
    '&:active': {
      backgroundColor: appTheme.colorAccentActive,
    },
    '&:hover': {
      backgroundColor: appTheme.colorAccentHover,
    },
  },
  normalIcon: {
    marginRight: appTheme.spacing1x,
    color: '#5c7080',
  },
  primaryIcon: {
    marginRight: appTheme.spacing1x,
  },
};

const stylesProvider = () => styles;

type RenderProps = Props & {
  classes: {
    button: string;
    normalButton: string;
    primaryButton: string;
    normalIcon: string;
    primaryIcon: string;
  };
};

const AppButton = withStyles(stylesProvider)(({ label, type, icon, style, onClick, classes }: RenderProps) => {
  let buttonClassName = '';
  let iconClassName = '';
  if (type === 'primary') {
    buttonClassName = `${classes.button} ${classes.primaryButton}`;
    iconClassName = classes.primaryIcon;
  } else {
    buttonClassName = `${classes.button} ${classes.normalButton}`;
    iconClassName = classes.normalIcon;
  }

  return (
    <Button variant="outlined" onClick={onClick} className={buttonClassName} style={style}>
      {icon && (
        <Icon className={iconClassName} fontSize="small">
          {icon}
        </Icon>
      )}
      {label}
    </Button>
  );
});

export default AppButton;
