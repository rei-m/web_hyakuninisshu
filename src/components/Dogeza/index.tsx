import * as React from 'react';
import styled from 'styled-components';
import * as dogeza from './dogeza_businessman.png';

const DogezaImg = styled.img`
  width: 200px;
`;

const Dogeza: React.SFC<React.ImgHTMLAttributes<{}>> = ({ alt, style }) => (
  <DogezaImg src={dogeza} alt={alt} style={style} />
);

export default Dogeza;
