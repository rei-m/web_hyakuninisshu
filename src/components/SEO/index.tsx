import * as React from 'react';
import Helmet from 'react-helmet';
import { graphql, StaticQuery } from 'gatsby';
import { SiteMetaData } from '@src/types';

export interface Props {
  description?: string;
  lang?: string;
  meta?: Array<{
    name: string;
    content: string;
  }>;
  keywords?: string[];
  title: string;
}

export interface DefaultSEOQueryData {
  site: {
    siteMetadata: SiteMetaData;
  };
  ogpImage: {
    publicURL: string;
  };
}

const SEO: React.FC<Props> = ({ description, lang, meta, keywords, title }) => {
  const checkedLang: string = lang ? lang : 'ja';
  const checkedKeywords: string[] = keywords ? keywords : [];
  const checkedMeta: Array<{
    name: string;
    content: string;
  }> = meta ? meta : [];

  return (
    <StaticQuery
      query={query}
      render={(data: DefaultSEOQueryData) => {
        const checkedTitle = title || data.site.siteMetadata.title;
        const checkedDescription = description || data.site.siteMetadata.description;
        return (
          <Helmet
            htmlAttributes={{
              lang: checkedLang,
            }}
            title={checkedTitle}
            meta={[
              {
                name: `description`,
                content: checkedDescription,
              },
              {
                property: `og:title`,
                content: checkedTitle,
              },
              {
                property: `og:description`,
                content: checkedDescription,
              },
              {
                property: `og:type`,
                content: `website`,
              },
              {
                property: `og:image`,
                content: `https://hyakuninanki.net${data.ogpImage.publicURL}`,
              },
              {
                name: `twitter:card`,
                content: `summary`,
              },
              {
                name: `twitter:creator`,
                content: data.site.siteMetadata.author,
              },
              {
                name: `twitter:title`,
                content: checkedTitle,
              },
              {
                name: `twitter:description`,
                content: checkedDescription,
              },
              {
                name: `twitter:image`,
                content: `https://hyakuninanki.net${data.ogpImage.publicURL}`,
              },
            ]
              .concat(
                checkedKeywords.length > 0
                  ? {
                      name: `keywords`,
                      content: checkedKeywords.join(`, `),
                    }
                  : []
              )
              .concat(checkedMeta)}
          />
        );
      }}
    />
  );
};

export default SEO;

const query = graphql`
  query DefaultSEOQuery {
    site {
      siteMetadata {
        title
        description
        author
      }
    }
    ogpImage: file(relativePath: { eq: "app-icon.png" }) {
      publicURL
    }
  }
`;
