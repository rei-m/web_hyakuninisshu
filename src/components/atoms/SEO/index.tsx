import React from 'react';
import Helmet from 'react-helmet';
import { useSEO } from '@src/hooks/staticQueries/useSEO';

export type Props = {
  description?: string;
  lang?: string;
  meta?: Array<{
    name: string;
    content: string;
  }>;
  keywords?: string[];
  title: string;
};

const SEO = ({ description, lang = 'ja', meta = [], keywords = [], title }: Props) => {
  const { site, ogpImage } = useSEO();
  const checkedTitle = title || site.siteMetadata.title;
  const checkedDescription = description || site.siteMetadata.description;
  return (
    <Helmet
      htmlAttributes={{
        lang,
      }}
      title={checkedTitle}
      meta={[
        {
          name: `description`,
          content: checkedDescription,
        },
        {
          property: `og:title`,
          content: checkedTitle,
        },
        {
          property: `og:description`,
          content: checkedDescription,
        },
        {
          property: `og:type`,
          content: `website`,
        },
        {
          property: `og:image`,
          content: `https://hyakuninanki.net${ogpImage.publicURL}`,
        },
        {
          name: `twitter:card`,
          content: `summary`,
        },
        {
          name: `twitter:creator`,
          content: site.siteMetadata.author,
        },
        {
          name: `twitter:title`,
          content: checkedTitle,
        },
        {
          name: `twitter:description`,
          content: checkedDescription,
        },
        {
          name: `twitter:image`,
          content: `https://hyakuninanki.net${ogpImage.publicURL}`,
        },
      ]
        .concat(
          keywords.length > 0
            ? {
                name: `keywords`,
                content: keywords.join(`, `),
              }
            : []
        )
        .concat(meta)}
    />
  );
};

export default SEO;
