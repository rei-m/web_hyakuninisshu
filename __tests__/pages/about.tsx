import * as ReactTestRenderer from 'react-test-renderer';
import * as React from 'react';
import About, { Props } from '@src/pages/about';
import { DefaultSEOQueryData } from '@src/components/SEO';
import { setUpQueryOnce } from '@test/helpers';

describe('/about', () => {
  let baseProps: Props;

  beforeEach(() => {
    setUpQueryOnce<DefaultSEOQueryData>({
      site: {
        siteMetadata: {
          title: 'about title',
          description: 'about description',
          author: '@rei-m',
        },
      },
    });
    baseProps = {
      data: {
        site: {
          siteMetadata: {
            description: 'about description',
          },
        },
      },
    };
  });

  it('should render component', () => {
    const renderer = ReactTestRenderer.create(<About {...baseProps} />);
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
