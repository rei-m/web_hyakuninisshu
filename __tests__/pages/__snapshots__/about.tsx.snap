// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/about should render component 1`] = `
Array [
  <header
    className="sc-bwzfXH geYBAS"
  >
    <div
      className="sc-bdVaJa flBAph"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <i
        className="material-icons sc-bxivhb jrBfKJ"
        color="#fff"
        data-test="back"
        onClick={[Function]}
      >
        arrow_back
      </i>
    </div>
    <h1
      className="sc-htpNat eSnwWs"
    >
      百人一首 - サイトについて -
    </h1>
  </header>,
  <div
    className="sc-htoDjs dfwcpY"
  >
    <div
      className="sc-fjdhpX gNyWJC"
    >
      <section
        className="sc-jzJRlG fsWdXU"
      >
        <h1
          className="sc-cSHVUG eJSQJf"
        >
          サイトについて
        </h1>
        <div
          className="sc-kAzzGY hkVRDm"
        >
          このサイトは百人一首を覚えるためのサイトです。4択のクイズ形式で繰り返し練習することで効率よく百人一首を覚えることが出来ます。
        </div>
      </section>
      <div
        className="sc-gZMcBi iXZiBx"
      >
        <div
          className="sc-gqjmRU cCYYjG"
        >
          この欄は広告枠です
        </div>
      </div>
      <section
        className="sc-jzJRlG fsWdXU"
      >
        <h1
          className="sc-cSHVUG eJSQJf"
        >
          使い方について
        </h1>
        <div
          className="sc-kAzzGY hkVRDm"
        >
          練習では様々な条件を組み合わせて出題範囲を指定できます。ご自身の習熟度に合わせて効率よく練習できます。 もう全部覚えたという方は腕試しに挑戦しましょう。百首通しでランダムに出題されます。 資料では百人一首の詳細な情報を閲覧できます。決まり字や歌の現代語訳などを確認することができます。
        </div>
      </section>
      <div
        className="sc-VigVT ieiGAu"
      >
        <div
          className="sc-jTzLTM ljXwZC"
        >
          この欄は広告枠です
        </div>
      </div>
      <section
        className="sc-jzJRlG fsWdXU"
      >
        <h1
          className="sc-cSHVUG eJSQJf"
        >
          アプリ版について
        </h1>
        <div
          className="sc-kAzzGY hkVRDm"
        >
          Androidのみリリースしています。こちらは一旦インストールしていただけばオフラインでも使えます。iOS版は未定ですが、要望があればがんばります。
        </div>
        <div
          className="sc-chPdSV kzZEeT"
        >
          <a
            href="https://play.google.com/store/apps/details?id=me.rei_m.hyakuninisshu&hl=ja&pcampaignid=MKT-Other-global-all-co-prtnr-py-PartBadge-Mar2515-1"
          >
            <img
              alt="Google Play で手に入れよう"
              src="https://play.google.com/intl/en_us/badges/images/generic/ja_badge_web_generic.png"
              style={
                Object {
                  "margin": "auto",
                  "width": 200,
                }
              }
            />
          </a>
        </div>
      </section>
      <section
        className="sc-jzJRlG fsWdXU"
      >
        <h1
          className="sc-cSHVUG eJSQJf"
        >
          運営者について
        </h1>
        <div
          className="sc-kAzzGY hkVRDm"
        >
          当サイトは
          <a
            href="https://twitter.com/rei_m"
            target="_blank"
          >
            @rei_m
          </a>
          が個人で運営しています。不具合や要望等あればTwitterまでご連絡ください。 アプリのレビューで読み上げ機能が欲しいというお声をいただくのですが、使用可能ないい音源があれば搭載します。
        </div>
      </section>
      <section
        className="sc-jzJRlG fsWdXU"
      >
        <h1
          className="sc-cSHVUG eJSQJf"
        >
          プライバシー ポリシー
        </h1>
        <div
          className="sc-kAzzGY hkVRDm"
        >
          サイトの利用状況の把握や広告配信のためにGoogle社が提供しているサービスを利用しております。詳しくは
          <a
            href="https://rei-m.github.io/app/policy/"
            target="_blank"
          >
            こちら
          </a>
          から確認ください。
        </div>
      </section>
    </div>
  </div>,
  <nav
    className="sc-bZQynM gfjWgY"
  >
    <span
      className="sc-gzVnrw klvbjl"
    >
      <div
        className="sc-bdVaJa flBAph"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          color="#fff"
          href="/training"
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <i
            className="material-icons sc-EHOje evPMQv"
          >
            create
          </i>
        </a>
      </div>
    </span>
    <span
      className="sc-gzVnrw klvbjl"
    >
      <div
        className="sc-bdVaJa flBAph"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          color="#fff"
          href="/exam"
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <i
            className="material-icons sc-EHOje gwxweT"
          >
            note
          </i>
        </a>
      </div>
    </span>
    <span
      className="sc-gzVnrw klvbjl"
    >
      <div
        className="sc-bdVaJa flBAph"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          color="#fff"
          href="/karutas"
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <i
            className="material-icons sc-EHOje kZpNpk"
          >
            library_books
          </i>
        </a>
      </div>
    </span>
    <span
      className="sc-gzVnrw klvbjl"
    >
      <div
        className="sc-bdVaJa flBAph"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          color="#fff"
          href="/about"
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <i
            className="material-icons sc-EHOje gbwtfN"
          >
            settings
          </i>
        </a>
      </div>
    </span>
  </nav>,
]
`;
